src/amg/azext_amg/custom.py

```diff
@@ -206,13 +211,13 @@ def update_grafana(cmd, grafana_name, api_key_and_service_account=None, determin
     return client.grafana.begin_create(resource_group_name, grafana_name, instance)
 
 
-def show_grafana(cmd, grafana_name, resource_group_name=None):
-    client = cf_amg(cmd.cli_ctx)
+def show_grafana(cmd, grafana_name, resource_group_name=None, subscription=None):
+    client = cf_amg(cmd.cli_ctx, subscription=subscription)
     return client.grafana.get(resource_group_name, grafana_name)
```

Comment:
Why the subscription is required only for show command?

Reply:
Good question. This is an internal argument for code reusing since "show_grafana" can be a util method. In `_params.py`, I use `c.ignore("subscription")` to be invisible to end users. Let me know if there are other better ways.